Codecademy/gamut
# feat(Form + GridForm): Add actual required optional + required form text

### Overview


Adds * Required to GridForm (and as a FormElement) + (optional) to labels of optional field. Also removed `showRequired` from GridForm because the accessibility text is mandatory 

Theres also some reorganizing of form elements into folders 


### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=47031-49129&m=dev)
- [x] Related to JIRA ticket: [gm-620]

#### Testing Instructions


 1. Check out GridForm page
 2. See the form additions
 3. Turn on screenreader 
 4. See that (optional) and * in form fields are not read

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Mono    | [Monolith PR](https://github.com/codecademy-engineering/mono/pull/6099)  | [Monolith Env] |
